src/test/java/com/project/springbootproject/controller/BookControllerTest.java

```diff
@@ -0,0 +1,301 @@
+        mockMvc.perform(delete("/books/{id}", id))
+                .andExpect(status().isUnauthorized());
+
+        // Then
+        Optional<Book> optionalBook = bookRepository.findById(id);
+        assertTrue(optionalBook.isPresent());
+    }
+
+    @WithMockUser(username = "adminF@ukr.net", roles = {"ADMIN"})
+    @Test
+    @Sql(scripts = "classpath:database/books/delete-all-books.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Update book by ID")
+    @Sql(scripts = "classpath:database/books/delete-all-books.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void updateBook_AdminRole_Success() throws Exception {
+        // Given
+        Book book = new Book();
+        book.setTitle("Sample Book");
+        book.setAuthor("Sample Author");
+        book.setIsbn("1234567890");
+        book.setPrice(BigDecimal.valueOf(19.99));
+        bookRepository.save(book);
+
+        Long id = book.getId();
+
+        BookRequestDto bookDto = new BookRequestDto()
+                .setTitle("Updated Title")
+                .setAuthor("Updated Author")
+                .setIsbn("Updated ISBN");
+        String jsonRequest = objectMapper.writeValueAsString(bookDto);
+
+        // When
+        mockMvc.perform(put("/books/{id}", id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(jsonRequest))
+                .andExpect(status().isNoContent());
+
+        // Then
+        Book updatedBook = bookRepository.findById(id)
+                .orElseThrow(() -> new AssertionError("Book not found"));
+
+        Assertions.assertEquals("Updated Title", updatedBook.getTitle());
+        Assertions.assertEquals("Updated Author", updatedBook.getAuthor());
+        Assertions.assertEquals("Updated ISBN", updatedBook.getIsbn());
+    }
+
+    @WithMockUser(username = "adminF@ukr.net", roles = {"ADMIN"})
+    @Test
+    @Sql(scripts = "classpath:database/books/delete-all-books.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Search books with parameters")
+    void searchBooksByParameters() throws Exception {
+        // Given
+        String[] authors = {"Author1", "Author2"};
+        String[] descriptions = {"Description1", "Description2"};
+        BookSearchParameters searchParameters = new BookSearchParameters(authors, descriptions);
+
+        BookRequestDto book1RequestDto = new BookRequestDto()
+                .setAuthor("Author1")
+                .setDescription("Description1")
+                .setTitle("Title1")
+                .setIsbn("ISBN1")
+                .setPrice(BigDecimal.valueOf(15.78));
+        BookRequestDto book2RequestDto = new BookRequestDto()
+                .setAuthor("Author2")
+                .setDescription("Description2")
+                .setTitle("Title2")
+                .setIsbn("ISBN2")
+                .setPrice(BigDecimal.valueOf(20.15));
```

Comment:
Similar issue for all test-classes. You create lot of dtos and models in test-code.
I recommend to create separate private methods for this and use them in test methods. This will make tests more short and readable